.flowconfig

```diff
@@ -85,6 +85,7 @@ munge_underscores=true
 
 module.name_mapper='^[./a-zA-Z0-9$_-]+\.\(bmp\|gif\|jpg\|jpeg\|png\|psd\|svg\|webp\|m4v\|mov\|mp4\|mpeg\|mpg\|webm\|aac\|aiff\|caf\|m4a\|mp3\|wav\|html\|pdf\)$' -> 'RelativeImageStub'
 module.name_mapper='@gutenberg' -> '<PROJECT_ROOT>/gutenberg'
+module.name_mapper='@wordpress/blocks' -> '<PROJECT_ROOT>/gutenberg/packages/blocks/src'
```

Comment:
I'm actually not sure we should name this `@wordpress` as blocks are actually `@gutenberg`'s. As a result of this observation we may also consider how other mappers are called, which I'm not sure of, but for Gutenberg blocks it sounds like it's definitely `@gutenberg`, wdyt?

Also, may I ask why do we set this here in Flow, and is not set in the module mapper plugin in `.babelrc` to indicate an alias? That's how I'd approached it, and would like to learn if that's a feasible path or not, and why this would be preferred, if the question first of all makes sense (i.e. can Flow be used for the same thing?)

Reply:
>I'm actually not sure we should name this @wordpress as blocks are actually @gutenberg's.

The package is actually called like that ([see its package definition here](https://github.com/WordPress/gutenberg/blob/master/packages/blocks/package.json#L2)) so, it makes sense to keep its original name.

> why do we set this here in Flow, and is not set in the module mapper plugin in .babelrc to indicate an alias

It's not an alias as far as the build pipeline is concerned. For Metro, `@wordpress/blocks` is a normal package and tries its best to resolve it. Metro being the smart tool it is, it actually _discovers_ the package automagically inside the `gutenberg/packages` folder.

We gradually have moved away from having to define aliases in Babel as the Gutenberg code has been npm-package-ized more and more. Previously, we had to "mock" those packages to mimic what Webpack is doing on the Gutenberg build side. But nowadays, the published Gutenberg packages can be easily picked up by Metro without alias tricks. Some packages (namely, the `editor` package` is still not published to NPM and that's why it's practically the only alias we have left in Babel.

But, `Flow` is not as smart as Metro. We need to give more direct instructions to Flow to go find the packages.

While on the subject, `Jest` (our testing tool) is also not as smart as Metro and that's why we have similar aliases in its `jest.config.js` file.

Reply:
Thanks for the detailed explanation about moving away from module resolver. I wonder why that wouldn't work then off the shelve for me (automagically as you say) when working on a similar PR in https://github.com/wordpress-mobile/gutenberg-mobile/tree/update/use-latest-gb-master

Reply:
> While on the subject, Jest (our testing tool) is also not as smart as Metro and that's why we have similar aliases in its jest.config.js file

Thanks for pointing this out as well, brings useful insight